### PR TITLE
Basic ratelimit for user login (including 2FA) and admin login

### PR DESCRIPTION
**Pull request by dani-garcia**
Might need to adjust the default rate limiting values and explain their use better.

Also needs to document their use in the .env template.

### CODE REVIEW

**Comment by BlackDex** [comment]
That looks very simple indeed. Will be a nice addition. Ill see if i can do some checks tomorrow :).

File: src/ratelimit.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            err_code!("Too many login requests", 429);
+        }
+    }
+}
+
+pub fn check_limit_admin(ip: &IpAddr) -> Result<(), Error> {
+    match LIMITER_ADMIN.check_key(ip) {
+        Ok(_) => Ok(()),
+        Err(_e) => {
+            err_code!("Too many admin requests", 429);
```
Comment (williamdes) [inline src/ratelimit.rs:35]: https://http.cat/

A bit of fun for everyone

**Review by williamdes — APPROVED** [review]

**Review by BlackDex — APPROVED** [review]

**Review by BlackDex — CHANGES_REQUESTED** [review]
All looks fine, but i'm missing the new config values in the `.env.template` file :).

File: src/config.rs [hunk 2]
```diff
@@ -511,6 +511,16 @@ make_config! {
 
         /// Allowed iframe ancestors (Know the risks!) |> Allows other domains to embed the web vault into an iframe, useful for embedding into secure intranets
         allowed_iframe_ancestors: String, true, def,    String::new();
+
+        /// Seconds between login requests |> Number of seconds, on average, between login requests before rate limiting kicks in. Note that this applies to both the login and the 2FA, so it's recommended to allow a burst size of at least 2
```
Comment (jjlin) [inline src/config.rs:515]: It would probably be helpful to clarify that this limit is per IP.

**Review by jjlin — CHANGES_REQUESTED** [review]

**Review by BlackDex — APPROVED** [review]

**Pull request merged**